File: MSSpotiAPI/src/main/java/com/ms/spotiapi/Models/Track.java
package com.ms.spotiapi.Models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.lang.Nullable;

import javax.persistence.*;
import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "TRACKS")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter

public class Track implements Serializable {
    @Id
    private String id;

    private String name;

    private int duration_ms;

    private String release_date;

    private String uri;

    @ManyToOne
    @JoinColumn(
            name = "tracks_album", nullable = false, referencedColumnName = "name"
    )
    private Album album;

    @ManyToMany
    @JoinTable(
            name = "tracks_artists",
            joinColumns = @JoinColumn(name = "track_name", referencedColumnName = "name"),
            inverseJoinColumns = @JoinColumn(name = "artist_name", referencedColumnName = "name")
    )
    private Set<Artist> artists = new HashSet<>();
}

File: LoadGenerator/Track.py
"""
████████╗██████╗░░█████╗░░█████╗░██╗░░██╗
╚══██╔══╝██╔══██╗██╔══██╗██╔══██╗██║░██╔╝
░░░██║░░░██████╔╝███████║██║░░╚═╝█████═╝░
░░░██║░░░██╔══██╗██╔══██║██║░░██╗██╔═██╗░
░░░██║░░░██║░░██║██║░░██║╚█████╔╝██║░╚██╗
░░░╚═╝░░░╚═╝░░╚═╝╚═╝░░╚═╝░╚════╝░╚═╝░░╚═╝
"""


class Track:
    def __init__(self, id, name, duration_ms, release_date, uri, album, artists):
        self.id = id
        self.name = name
        self.duration_ms = duration_ms
        self.release_date = release_date
        self.uri = uri
        self.album = album
        self.artists = artists

    def to_dict(self):
        artists = [artist.to_dict() for artist in self.artists]
        albums = [album.to_dict() for album in self.albums]
        return {"id": self.id, "name": self.name, "duration_ms": self.duration_ms, "release_date": self.release_date,
                "uri": self.uri, "album": self.album, "albums": albums, "artists": artists}
